### PR TITLE
Fix issue #3 - Failed to deserialize boolean literal and to primitive boolean properties

### PR DESCRIPTION
**Pull request by moghaddam**
Sorry, I didn't notice that GitHub creates an issue for each pull requests, so I first created an issue for my first two pull requests. This is to fix the issue #3

### CODE REVIEW

**Comment by m0mus** [comment]
Looks good to me. @bravehorsie please review and merge it with the main repo if ok.

File: src/main/java/org/eclipse/persistence/json/bind/internal/serializer/BooleanTypeDeserializer.java [hunk 1]
```diff
@@ -28,6 +34,21 @@ public BooleanTypeDeserializer(JsonBindingModel model) {
     }
 
     @Override
+    public Boolean deserialize(JsonParser parser, DeserializationContext ctx, Type rtType) {
```
Comment (bravehorsie) [inline src/main/java/org/eclipse/persistence/json/bind/internal/serializer/BooleanTypeDeserializer.java]: Unfortunatelly JsonParser API doesn't have any getBooleanValue method so this is the only possible solution.

File: src/main/java/org/eclipse/persistence/json/bind/internal/serializer/BooleanTypeDeserializer.java [hunk 2]
```diff
@@ -28,6 +34,21 @@ public BooleanTypeDeserializer(JsonBindingModel model) {
...
+            case VALUE_FALSE:
+                return Boolean.FALSE;
+            case VALUE_STRING:
+                return Boolean.parseBoolean(parser.getString());
+            default:
+                throw new JsonbException(Messages.getMessage(MessageKeys.INTERNAL_ERROR, "Unknown JSON value: " + event));
+        }
+    }
+
+    @Override
```
Comment (bravehorsie) [inline src/main/java/org/eclipse/persistence/json/bind/internal/serializer/BooleanTypeDeserializer.java]: Looks like we don't need the second deserialize method from AbstractValueTypeDeserializer here, please change "return Boolean.parseBoolean(jsonValue);" with throw new UnsupportedOperationException and add TODO: fix API.

Comment (moghaddam) [inline src/main/java/org/eclipse/persistence/json/bind/internal/serializer/BooleanTypeDeserializer.java]: You're right. I updated the second implementation of the deserialize method as you said.

**Pull request merged**